File: modul_praktikum_python/contoh_code/pascal/record_upah.pas
uses crt; 
type 
  Tpegawai=record 
           Nama : string[20]; 
           Gol  : char; 
           JmlAnak : integer; 
           end; 
var 
  pegawai:tpegawai; 
  Upahkotor,upahbersih : real; 
  PersenTunjangan:real; 
Begin 
  Write(‘Nama        : ’);readln(pegawai.Nama); 
  Write(‘Gol (A - D) : ’);readln(pegawai.Gol); 
  Write(‘Jumlah Anak : ’);readln(pegawai.JmlAnak); 

  Case pegawai.Gol of 
  ‘A’ : Upahkotor:=1000000; 
  ‘B’ : Upahkotor:=800000; 
  ‘C’ : Upahkotor:=600000; 
  ‘D’ : Upahkotor:=400000; 
  end;   

  if(pegawai.JmlAnak>2)then  
    PersenTunjangan := 0.3   
  else 
   PersenTunjangan := 0.2; 

  Upahbersih:= upahkotor – (upahkotor*persentunjangan); 
  writeln(‘Upah        : ’,upahbersih); 
end.

File: modul_praktikum_python/contoh_code/python/record_upah.py
"""Record pegawai: upah kotor menurut gol dan tunjangan anak."""

from dataclasses import dataclass


@dataclass
class Pegawai:
    nama: str = ""
    gol: str = " "
    jml_anak: int = 0


def main() -> None:
    pegawai = Pegawai()
    pegawai.nama = input("Nama        : ")
    pegawai.gol = input("Gol (A - D) : ").strip().upper()[:1]
    pegawai.jml_anak = int(input("Jumlah Anak : "))

    if pegawai.gol == "A":
        upah_kotor = 1_000_000.0
    elif pegawai.gol == "B":
        upah_kotor = 800_000.0
    elif pegawai.gol == "C":
        upah_kotor = 600_000.0
    elif pegawai.gol == "D":
        upah_kotor = 400_000.0
    else:
        upah_kotor = 0.0

    if pegawai.jml_anak > 2:
        persen_tunjangan = 0.3
    else:
        persen_tunjangan = 0.2

    # Pascal: typo persentunjangan - gunakan persen_tunjangan
    upah_bersih = upah_kotor - (upah_kotor * persen_tunjangan)
    print("Upah        :", upah_bersih)


if __name__ == "__main__":
    main()
